feat: support named pipe listen, connect and open

fraidev opened the pull request:
Co-authored-by:  Cyan Changes <contact@cyans.me>

Add support to Pipe.prototype.listen, Pipe.prototype.connect for windows with named pipes. 
Also support to Pipe.prototype.open for unix.

Alternative of https://github.com/denoland/deno/pull/29308, but without `Deno:pipe`.
Only with node compatibilities.


Solution for:
https://github.com/denoland/deno/issues/25867
https://github.com/denoland/deno/issues/28332
https://github.com/denoland/deno/issues/31032


Maybe related:
https://github.com/denoland/deno/issues/10244


I tested the Nuxt and Nx, they are working fine on Windows now.
I also tested node-pty, and it's working on Unix.

bartlomieju commented on ext/net/ops_win_pipe.rs:
```diff
@@ -0,0 +1,150 @@
+#[serde]
+pub fn op_pipe_open(
+  state: &mut OpState,
+  #[serde] args: ListenArgs,
```
A bit nitpicky, but I'd suggest to "unfurl" these argument and just specify all fields directly and not relay on serde and GC

bartlomieju commented on ext/net/ops_win_pipe.rs:
```diff
@@ -0,0 +1,150 @@
+) -> Result<ResourceId, NetError> {
+  let permissions = state.borrow_mut::<PermissionsContainer>();
+
+  let api_call_expr = format!("{}()", api_name);
```
Ditto

fraidev replied:
fixed!

bartlomieju commented on ext/node/polyfills/internal_binding/pipe_wrap.ts:
```diff
@@ -368,6 +587,15 @@ export class Pipe extends ConnectionWrap {
     if (this.provider === providerType.PIPESERVERWRAP) {
+      if (this.#serverPipeRid !== undefined) {
+        try {
+          core.close(this.#serverPipeRid);
```
This can be `core.tryClose` without `try/close`

CyanChanges commented on ext/node/polyfills/internal_binding/pipe_wrap.ts:
```diff
@@ -98,12 +101,12 @@ class FileStreamConn {
         ) {
           // Wait a bit before retrying to avoid busy-looping
-          await new Promise((resolve) => setTimeout(resolve, 10));
+          await delay(10);
```
Is delaying a good idea?

fraidev commented:
> That's great. It's still under review, but would it be better to include @CyanChanges as a co-author?

Sure, I squashed all commits and added him.

fraidev commented:
> > Sure, I squashed all commits and added him.
> 
> While it's not wrong to specify an author for each commit, adding it to the PR summary is probably sufficient like below.👍
> 
> * [refactor: add deno_config crate back to cli #29740](https://github.com/denoland/deno/pull/29740)

Oh, I didn't know that, thanks. I also add it in the PR description.
